hibernate-reactive-core/src/test/java/org/hibernate/reactive/MutinySessionTest.java

```diff
@@ -0,0 +1,250 @@
+package org.hibernate.reactive;
+
+import io.smallrye.mutiny.Uni;
+import io.vertx.ext.unit.TestContext;
+import org.hibernate.cfg.Configuration;
+import org.junit.Test;
+
+import javax.persistence.Entity;
+import javax.persistence.Id;
+import javax.persistence.Table;
+import java.util.Objects;
+
+public class MutinySessionTest extends BaseMutinyTest {
+
+	@Override
+	protected Configuration constructConfiguration() {
+		Configuration configuration = super.constructConfiguration();
+		configuration.addAnnotatedClass( GuineaPig.class );
+		return configuration;
+	}
+
+	private Uni<Integer> populateDB() {
+		return connection().flatMap(
+				conn -> Uni.createFrom().completionStage( conn.update( "INSERT INTO Pig (id, name) VALUES (5, 'Aloi')" ) )
+		);
+	}
+//
+//	private Uni<Integer> cleanDB() {
+//		return connection().flatMap( conn -> Uni.createFrom().completionStage( conn.update( "DELETE FROM Pig" ) ) );
+//	}
+
+	private Uni<String> selectNameFromId(Integer id) {
+		return connection().flatMap(
+				connection -> Uni.createFrom().completionStage(
+						connection.select(
```

Comment:
Is it normal to have to use `CompletionStage` for `select`?

Reply:
There is no Mutiny API for `ReactiveConnection`. It's an internal API, not supposed to be visible to the program. (You have to be an expert in Hibernate `Service`s to even get a reference to it.)

Reply:
OK, fair.